src/Squirrel/Utility.cs

```diff
@@ -388,6 +406,12 @@ public static async Task DeleteDirectory(string directoryPath)
             } catch (Exception ex) {
                 var message = String.Format("DeleteDirectory: could not delete - {0}", directoryPath);
                 Log().ErrorException(message, ex);
+                if (moveFirst) {
+                    // move back to original name so we can try again on the next run
+                    Directory.Move(directoryPath, originalPath);
```

Comment:
interesting. so `Move()` throws and then we're left with a partially moved directory that this second call restores, presumably only touching files that weren't locked & unable to be moved the first time?


and then how does this manifest to the user when they run `Setup.exe`, as a dialog as i think was mentioned somewhere?

Reply:
when DeleteDirectory throws during the install action, you get a generic error modal with a button that opens the setup log file that contains the actual exception

if the initial `Move()` throws, nothing will have been modified (which is the goal here). if this `Move()` fails we have a stickier situation as it means we have failed to fully delete the directory and we weren't able to rename it back to the original name. So we will effectively be leaving some garbage behind in the user's appdata directory in this case. The hope is that this will be a rare failure in practice since the whole point of doing the initial `Move()` is that it will fail if any of the children are locked and so the recursive delete will hopefully succeed in full. Even if the delete fails, the hope is that if we were able to rename the directory once, we'll be able to do it again.